src/components/user/login.js

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from 'react';
+import '../../stylesheets/login.css';
+import { useNavigate } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { toast } from 'react-toastify';
+import { loginUser } from '../../redux/user/userSlice';
+
+const Login = ({ onClose }) => {
+  const { isLoggedIn } = useSelector((state) => state.users);
+  const showModal = useSelector((state) => state.modal.showModal);
+
+  const dispatch = useDispatch();
+
+  const [loginEmail, setLoginEmail] = useState('');
+  const [loginPassword, setLoginPassword] = useState('');
+  const navigate = useNavigate();
+
+  const resetData = () => {
+    setLoginEmail('');
+    setLoginPassword('');
+  };
+
+  useEffect(() => {
+    if (loginEmail !== '' && loginPassword !== '') {
+      const data = {
+        user: {
+          email: loginEmail,
+          password: loginPassword,
+        },
+      };
+      try {
+        dispatch(loginUser(data));
+        if (isLoggedIn) {
+          navigate('/mainpage');
+          resetData();
```

Comment:
There are a handful of weird errors when logging in. If one tries to enter with a user, logs out, and tries to log in again, the request fires off to log a user in regardless if they've finished writing their password. Here's the error I see on console after. 
![image](https://github.com/leloufadel/front-end-capstone/assets/91759364/4cf314b8-9ab5-49de-8c8e-18c3532b4e59)

Here are two instances of the error ocurring
(apologies for the audio on the video, forgot to turn off mic)

https://github.com/leloufadel/front-end-capstone/assets/91759364/7503b0d3-f16c-4ae2-9bb4-80e37dcd6517

https://github.com/leloufadel/front-end-capstone/assets/91759364/71a37421-d20a-4a0e-9313-540192f4aae4

Please look into this.

Reply:
Ready.